atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchEventCacheImpl.java

```diff
@@ -107,7 +93,7 @@ public SpanningCommitUpdate getSpanningCommitUpdate(long startTs) {
                 maybeCommitInfo.isPresent()
                         && startVersion.isPresent()
                         && eventLog.getLatestKnownVersion().isPresent(),
```

Comment:
I guess you could extract a method to avoid throwing here for `getCommitUpdate` if latest known version is not present

Reply:
I could, but actually I don't want to, since if you don't have a current version, you most definitely do not have any of the other information present.